scipy/spatial/transform/rotation.py

```diff
@@ -8,7 +8,102 @@
 AXIS_TO_IND = {'x': 0, 'y': 1, 'z': 2}
 
 
-def _make_elementary_quat(axis, angles):
+def elementary_basis_vector(axis):
+    b = np.zeros(3)
+    b[AXIS_TO_IND[axis]] = 1
+    return b
+
+
+def compute_euler_from_dcm(dcm, seq, extrinsic=False):
+    # The algorithm assumes intrinsic frame transformations. For representation
+    # the paper uses transformation matrices, which are transpose of the
+    # direction cosine matrices used by our Rotation class.
+    # Adapt the algorithm for our case by
+    # 1. Transposing our dcm representation
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+
+    if extrinsic:
+        seq = seq[::-1]
+
+    if dcm.ndim == 2:
+        dcm = dcm[None, :, :]
+    num_rotations = dcm.shape[0]
+
+    # Step 0
+    # Algorithms dcms are transpose of ours
+    dcm = np.transpose(dcm, (0, 2, 1))
+    # Algorithm assumes axes as column vectors, here we use 1D vectors
+    n1 = elementary_basis_vector(seq[0])
+    n2 = elementary_basis_vector(seq[1])
+    n3 = elementary_basis_vector(seq[2])
+
+    # Step 2
+    sl = np.dot(np.cross(n1, n2), n3)
+    cl = np.dot(n1, n3)
+
+    norm = sl**2 + cl**2
```

Comment:
You're right. I'm not sure about the general case presented in the paper, but since we're only dealing with axes parallel to the coordinate axes, norm will always be one. Removing it.